[Duplicate of #5919]Data representation using brython (#6156)
section-engineering-education/engineering-education: content/articles/data-representation-using-brython/index.md

```javascript
<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.10.4/brython.js" integrity="sha512-Dr8IR/2fk/pBmngFLduWN5Ys/t7lcPxGgLCulHdTdgDrDRnGb+ANMxGXq+DKVN87jom+N3BrzzI476Pf98RReA==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.10.4/brython_stdlib.js" integrity="sha512-kMRN6F4Yq4sNLbPG2lH3EO9n776JHHZub+UWogDxVjh9uTnoVo3wtN/rnQD4C4/AZtqI2zQdvdouGAAxOGwNeA==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```

This is by far the easiest method to import external files in HTML. In cases of you being offline, it is a huge disadvantage.

Any of the methods above is efficient to integrate Brython JavaScript and CSS into our project. Now, We can go to the next step.

#### Setting up the environment
The following steps will contain how the HTML file should be set up for our pie chart and scatterplot. Here is a step by step method to achieve this:

##### Installing the Highcharts module.
To install the Highcharts module, use this code:

```npm
npm install highcharts
```

> 
> Doing this will download some important Highchart modules such as `highchart.js` and `exporting.js`. The exporting module allows you to download the charts in different file formats like PDF, PNG, SVG vector images, OR JPG.
> 
> We will need to add `Jquery, exporting and highchart modules` to our HTML boilerplate. To do this, we'll refer to their respective file using the script tag below:
> 

Reviewer: ```suggestion

Using the command above, we will download some important Highchart modules such as `highchart.js` and `exporting.js`. The `exporting.js` allows us to download the charts in different file formats like PDF, PNG, vector images(SVG), and JPG.

In our HTML boilerplate, we will add the  `Jquery` script, the `exporting.js` file, and the `highchart.js` module. Use the following code snippet:

```

                    ["Others", 0.5],
                ],
            }
        ],
    }
)
```
The code above plots the pie chart using the following methods:
- `from browser import window`: imports the window module from, thus allowing Brython access to Javascript object.
- `plotBackgroundColor`: It plots the background color for the chart area.
- `plotBorderWidth`: It plots the width of the chart area in pixels.
- `plotShadow`: This method takes in boolean values and adds a drop shadow to the area. 
- `renderTo`: This method is like `document.getElementBy` in javascript. It will take the `id` of the `div` tag in our HTML code.
- `tooltip`: This object contains the details in the small box when we hover on our chart.
- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.

When we execute this code, our pie chart is as shown below:

![piechart1](engineering-education/data-representation-with-react-and-ant-design/piechart1.png) 

![piechart2](engineering-education/data-representation-with-react-and-ant-design/piechart2.png)

> 

Reviewer: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order. 

 For example, in beginning, you have said  "Having created our ` chart.py file`"... but there is no `chart.py` file we have created. You have created the `chart.py` file later on in the article.  First, create the `chart.py' file. Then you can import/add this file into your HTML boilerplate.  When creating the `chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the `chart.py`,  `brython.js`, `brython_stdlib.js`, and other files. I believe this HTML file is the final part that holds all the imported files. 

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

> #### Creating a scatterplot using Brython.
> A scatterplot tests the relationship between two factors. To achieve this, we will put in place the HTML file used for the pie chart. The only difference is the code in the chart.py file.
> 

Reviewer: ```suggestion
#### Creating a scatterplot using Brython.
A scatterplot tests the relationship between two factors. To achieve this, we will use the HTML file that we have previously used to create the pie chart. We only need to make some changes to the created chart.py file.

```

> 

Reviewer: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order.

For example, in beginning, you have said "Having created our chart.py file"... but there is no chart.py file we have created. You have created the chart.py file later on in the article. First, create the chart.py' file. Then you can import/add this file into your HTML boilerplate. When creating the chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the chart.py, brython.js, brython_stdlib.js, and other files. I believe this HTML file is the final part that holds all the imported files.

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

> 

Reviewer: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order.

For example, in beginning, you have said "Having created our chart.py file"... but there is no chart.py file we have created. You have created the chart.py file later on in the article. First, create the chart.py' file. Then you can import/add this file into your HTML boilerplate. When creating the chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the chart.py, brython.js, brython_stdlib.js, and other files. I believe this HTML file is the final part that holds all the imported files.

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

Author: effected this changes

> ##### Chart.py file

Reviewer: ```suggestion
##### Making changes to the Chart.py file
```

> The `chart.py` file for the scatterplot is as below:

Reviewer: ```suggestion
The new `chart.py` file that we will use to plot the scatterplot is shown below:
```

```python
from browser import window

b_highchart = window.Highcharts.Chart.new
b_highchart(
    {
        "chart": {"type": "scatter", "zoomType": "xy", "renderTo": "container"},
        "title": {
            "text": "scatterplot of the height compared to the weight of 53 Individuals by Race."
        },
        "xAxis": {"title": {"text": "Height (cm)"}},
        "yAxis": {"title": {"text": "Weight (kg)"}},
        "plotOptions": {
            "scatter": {
                "marker": {
                    "radius": 5,
                    "states": {
                        "hover": {"enabled": True, "lineColor": "rgb(100,100,100)"}
                    },
                },
                "states": {"hover": {"marker": {"enabled": False}}},
                "tooltip": {
                    "headerFormat": "<b>{series.name}</b><br>",
                    "pointFormat": "{point.x} cm, {point.y} kg",
                },
            }
        },

>         "series": [

Reviewer: You can add the function of this `series` attribute when explaining this code snippet.

Author: I am trying to avoid repetitions. I already explained it in the piechart section

                    [108, 143],
                    [109, 92],
                    [108, 158],
                    [110, 102],
                    [79, 149],
                    [92, 141],
                    [86, 190],
                    [71, 99],
                    [129, 108],
                    [123, 116],
                    [89, 86],
                    [132, 181],
                    [137, 190],
                    [132, 143],
                ],
            },
            {
                "name": "Hispanic",
                "color": "green",
                "data": [
                    [156, 177],
                    [177, 83],
                    [180, 145],
                    [122, 106],
                    [100, 147],
                    [155, 72],
                    [85, 81],
                    [90, 122],
                    [107, 161],
                    [73, 168],
                    [100, 104],
                    [117, 82],
                    [153, 71],
                    [88, 74],
                    [171, 117],
                    [83, 185],
                    [125, 136],
                ],
            },
        ],
    }
)
```

> The code above shows the relation between weight and height using a sample of 53 individuals of different races.
> 
> When we execute this code, the scatterplot is as shown below:
> 
> ![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)
> 
> The code above plots the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
> 
> - `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
> - `zoomType`: This method specifies the zoom dimension when we move the mouse.
> - `radius`: This specifies the horizontal distance between the two closest points in a series.
> - `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
> - `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.
> 

Reviewer: ```suggestion
The code snippet above shows the relation between weight and height using a sample of 53 individuals of different races.

The code is used to plot the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
- `zoomType`: This method specifies the zoom dimension when we move the mouse.
- `radius`: This specifies the horizontal distance between the two closest points in a series.
- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.

When we execute this code, the scatterplot is as shown below:

![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)
```

### Conclusion
The process of data representation is a delightful journey. Like all journeys, it must come to an end.  Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).

> 

Reviewer: ```suggestion
Happy coding!
---
Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
```

                        "y": 10.0,
                        "sliced": True,
                        "selected": True,
                    },
                    ["Islam", 43.5],
                    ["Others", 0.5],
                ],
            }
        ],
    }
)
```
The code above plots the pie chart using the following methods:
- `from browser import window`: imports the window module from, thus allowing Brython access to Javascript object.
- `plotBackgroundColor`: It plots the background color for the chart area.
- `plotBorderWidth`: It plots the width of the chart area in pixels.
- `plotShadow`: This method takes in boolean values and adds a drop shadow to the area. 
- `renderTo`: This method is like `document.getElementBy` in javascript. It will take the `id` of the `div` tag in our HTML code.
- `tooltip`: This object contains the details in the small box when we hover on our chart.
- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.

When we execute this code, our pie chart is as shown below:

> 
> ![piechart1](engineering-education/data-representation-with-react-and-ant-design/piechart1.png) 
> 
> ![piechart2](engineering-education/data-representation-with-react-and-ant-design/piechart2.png)
> 

Reviewer: ```suggestion

![piechart1](engineering-education/data-representation-using-brython/piechart1.png) 

![piechart2](engineering-education/data-representation-using-brython/piechart2.png)

```

> 
> ![piechart1](engineering-education/data-representation-with-react-and-ant-design/piechart1.png) 
> 
> ![piechart2](engineering-education/data-representation-with-react-and-ant-design/piechart2.png)
> 

Reviewer: ```suggestion

![piechart1](engineering-education/data-representation-using-brython/piechart1.png) 

![piechart2](engineering-education/data-representation-using-brython/piechart2.png)

```

Reviewer: Why is there a need for using 2 piechart diagrams that show the same information? Please remove the redundant one. And rename the remaining one from `piechart1` to `piechart`. Also, do that the image name.

                    [100, 104],
                    [117, 82],
                    [153, 71],
                    [88, 74],
                    [171, 117],
                    [83, 185],
                    [125, 136],
                ],
            },
        ],
    }
)
```

The code snippet above shows the relation between weight and height using a sample of 53 individuals of different races.

The code is used to plot the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
- `zoomType`: This method specifies the zoom dimension when we move the mouse.
- `radius`: This specifies the horizontal distance between the two closest points in a series.
- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.

When we execute this code, the scatterplot is as shown below:

> ![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)

Reviewer: ```suggestion
![Scatterplot](engineering-education/data-representation-using-brython/scatterplot.png)
```

> ### Conclusion
> The process of data representation is a delightful journey. Like all journeys, it must come to an end.  Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).
> Happy coding!
> ---
> Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
> ### Further Reading
> For further reading on Brython, read this [documentation](https://brython.info/static_doc/en/intro.html).

Reviewer: ```suggestion
### Conclusion
The process of data representation is a delightful journey. Like all journeys, it must come to an end. Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. 

To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).

### Further Reading
For further reading on Brython, read this [documentation](https://brython.info/static_doc/en/intro.html).

I hope you find this article helpful.

Happy coding!

---
Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
```

                        "y": 10.0,
                        "sliced": True,
                        "selected": True,
                    },
                    ["Islam", 43.5],
                    ["Others", 0.5],
                ],
            }
        ],
    }
)
```
The code above plots the pie chart using the following methods and attributes:
- `from browser import window`: imports the window module from, thus allowing Brython access to Javascript object.
- `plotBackgroundColor`: It plots the background color for the chart area.
- `plotBorderWidth`: It plots the width of the chart area in pixels.
- `plotShadow`: This method takes in boolean values and adds a drop shadow to the area. 
- `renderTo`: This method is like `document.getElementBy` in javascript. It will take the `id` of the `div` tag in our HTML code.
- `tooltip`: This object contains the details in the small box when we hover on our chart.
- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.

When we execute this code, our pie chart is as shown below:

> ![piechart1](engineering-education/data-representation-using-brython/piechart1.png) 

Reviewer: ```suggestion
![Piechart](engineering-education/data-representation-using-brython/piechart1.png) 
```

Reviewer: Make that change and your article will be ready for final review.
